### DISCUSSION TITLE
Inconsistent numeric types?

### DISCUSSION THREAD

**Discussion by olalonde** [image-rs/image#1864] [General]
https://docs.rs/image/latest/image/imageops/fn.overlay.html

```rust
pub fn overlay<I, J>(bottom: &mut I, top: &J, x: i64, y: i64)
```

https://docs.rs/image/latest/image/fn.image_dimensions.html

```rust
pub fn image_dimensions<P>(path: P) -> ImageResult<(u32, u32)>
```

If an image has max `u32` dimension why does overlay take `i64` arguments? I just found that curious.

**Comment by fintelia** [comment]
The choice of using `i64` for the coordinates enables specifying positions that only partially overlap the bottom image. So for instance, the point (-1, -1) is a totally valid input for that function.
